File: trunk/Projects/CSharp/APISample/EntryPoint.cs
using System;
using CodeSmith.Engine;
using SchemaExplorer;

namespace APISample {
    public class EntryPoint {
        [STAThread]
        public static void Main() {
            string path = System.IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\StoredProcedures.cst");
            var engine = new TemplateEngine(new DefaultEngineHost(System.IO.Path.GetDirectoryName(path)));

            CompileTemplateResult result = engine.Compile(path);
            if (result.Errors.Count == 0) {
                var database = new DatabaseSchema(new SqlSchemaProvider(), @"Server=.;Database=PetShop;Integrated Security=True;");
                TableSchema table = database.Tables["Inventory"];

                CodeTemplate template = result.CreateTemplateInstance();
                template.SetProperty("SourceTable", table);
                template.SetProperty("IncludeDrop", false);
                template.SetProperty("InsertPrefix", "Insert");

                template.Render(Console.Out);
            } else {
                foreach (var error in result.Errors)
                    Console.Error.WriteLine(error.ToString());
            }

            Console.WriteLine("\r\nPress any key to continue.");
            Console.ReadKey();
        }
    }
}

File: Projects/VisualBasic/APISample/EntryPoint.vb
Imports System
Imports CodeSmith.Engine
Imports SchemaExplorer

Module EntryPoint
    Sub Main()
        Dim path As String = IO.Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "..\..\StoredProcedures.cst")
        Dim engine As New TemplateEngine(New DefaultEngineHost(IO.Path.GetDirectoryName(path)))

        Dim result As CompileTemplateResult = engine.Compile(path)
        If result.Errors.Count = 0 Then
            Dim database As New DatabaseSchema(New SqlSchemaProvider(), "Server=.;Database=PetShop;Integrated Security=True;")
            Dim table As TableSchema = database.Tables("Inventory")

            Dim template As CodeTemplate = result.CreateTemplateInstance()
            template.SetProperty("SourceTable", table)
            template.SetProperty("IncludeDrop", False)
            template.SetProperty("InsertPrefix", "Insert")

            template.Render(Console.Out)
        Else
            For Each e As Object In result.Errors
                Console.Error.WriteLine(e.ToString())
            Next
        End If

        Console.WriteLine(Environment.NewLine + "Please press any key to continue.")
        Console.ReadKey()
    End Sub
End Module